skstore/ts/src/skstore_api.ts

```diff
@@ -0,0 +1,588 @@
+  SKIP_SKStore_asyncLazy(ctx: ptr, name: ptr, paramsFn: int, lazyFn: int): ptr;
+  SKIP_SKStore_lazy(ctx: ptr, name: ptr, lazyFn: int): ptr;
+  SKIP_SKStore_fromSkdb(ctx: ptr, table: ptr, name: ptr, fnPtr: int): ptr;
+  SKIP_SKStore_multimap(ctx: ptr, name: ptr, mappings: ptr): ptr;
+  SKIP_SKStore_asyncResult(
+    callId: ptr,
+    name: ptr,
+    key: ptr,
+    param: ptr,
+    value: ptr,
+  ): number;
+
+  SKIP_SKStore_multimapReduce(
+    ctx: ptr,
+    name: ptr,
+    mappings: ptr,
+    accumulator: int,
+    accInit: ptr,
+  ): ptr;
+}
+
+export interface SKJSON extends Shared {
+  importJSON: (value: ptr, copy?: boolean) => any;
+  exportJSON: <T>(v: T) => ptr;
+  importOptJSON: (value: Opt<ptr>, copy?: boolean) => any;
+  importString: (v: ptr) => string;
+  exportString: (v: string) => ptr;
+  runWithGC: <T>(fn: () => T) => T;
+}
+
+export interface Context {
+  mapReduce: <
+    K extends TJSON,
+    V extends TJSON,
+    K2 extends TJSON,
+    V2 extends TJSON,
+    V3 extends TJSON,
+  >(
+    eagerHdl: string,
+    metadata: Metadata,
+    mapper: Mapper<K, V, K2, V2>,
+    accumulator: Accumulator<V2, V3>,
+  ) => string;
+
+  map: <K extends TJSON, V extends TJSON, K2 extends TJSON, V2 extends TJSON>(
+    eagerHdl: string,
+    metadata: Metadata,
+    compute: Mapper<K, V, K2, V2>,
+  ) => string;
+  input: <K extends TJSON, V extends TJSON>(
+    name: string,
+    value: [K, V][],
+  ) => string;
+  lazy: <K extends TJSON, V extends TJSON>(
+    metadata: Metadata,
+    compute: Lazy<K, V>,
+  ) => string;
+  asyncLazy: <
+    K extends TJSON,
+    V extends TJSON,
+    P extends TJSON,
+    M extends TJSON,
+  >(
+    metadata: Metadata,
+    get: (key: K) => P,
+    call: (key: K, params: P) => Promise<AValue<V, M>>,
+  ) => string;
+  get: <K, V>(eagerHdl: string, key: K) => V;
+  getFromTable: <K, R>(table: string, key: K, index?: string) => R[];
+  maybeGet: <K, V>(eagerHdl: string, key: K) => Opt<V>;
+  getLazy: <K, V>(lazyHdl: string, key: K) => V;
+  getSelf: <K, V>(lazyHdl: ptr, key: K) => V;
+
+  size: (eagerHdl: string) => number;
+
+  mapFromSkdb: <R extends TJSON, K extends TJSON, V extends TJSON>(
+    table: string,
+    metadata: Metadata,
+    mapper: EntryMapper<R, K, V>,
+  ) => string;
+  mapToSkdb: <R extends TJSON, K extends TJSON, V extends TJSON>(
+    eagerHdl: string,
+    table: string,
+    mapper: OutputMapper<R, K, V>,
+  ) => void;
+
+  multimap<
+    K1 extends TJSON,
+    V1 extends TJSON,
+    K2 extends TJSON,
+    V2 extends TJSON,
+  >(
+    metadata: Metadata,
+    mappings: Mapping<K1, V1, K2, V2>[],
+  ): string;
+  multimapReduce<
+    K1 extends TJSON,
+    V1 extends TJSON,
+    K2 extends TJSON,
+    V2 extends TJSON,
+    V3 extends TJSON,
+  >(
+    metadata: Metadata,
+    mappings: Mapping<K1, V1, K2, V2>[],
+    accumulator: Accumulator<V2, V3>,
+  ): string;
+
+  noref: () => Context;
+  notify?: () => void;
+}
+
+/**
+ * Collect function file line colum
```

Comment:
This docstring doesn't say much! (and I don't understand the comment on line 580, either)

- why is this one function in this file being implemented?  does it really belong here in the "`skstore_api`"?

- what assumptions are being made with the regex (and magic constant 2 added to the offset) that might break if some detail changes with the underlying stack representation?